### PR TITLE
Add qesap regression test on Azure: 15-SP5 PAYG

### PR DESCRIPTION
**Pull request by lilyeyes**
Add qesap regression tests: saptune enabled and sapconf enabled 
 
TEAM-8000 - [OSD] qesap regression test on Azure using 15-SP5 PAYG images

- Related ticket: https://jira.suse.com/browse/TEAM-8000
- Needles: NA
- Verification run:
  saptune enabled: https://openqa.suse.de/tests/11216912 (passed)
  sapconf enabled: https://openqa.suse.de/tests/11216913 (failed on ansible part as it doesn't work on 15-SP5 PAYG images yet)

### CODE REVIEW

**Comment by lpalovsky** [comment]
LGTM

**Review by a-kpappas — APPROVED** [review]

**Review by alvarocarvajald — APPROVED** [review]
LGTM

File: tests/sles4sap/qesapdeployment/configure.pm [hunk 1]
```diff
@@ -34,6 +34,7 @@ sub run {
     }
     $variables{OS_OWNER} = get_var('QESAP_CLUSTER_OS_OWNER', 'amazon') if check_var('PUBLIC_CLOUD_PROVIDER', 'EC2');
 
+    $variables{USE_SAPCONF} = get_required_var('USE_SAPCONF');
```
Comment (mpagot) [inline tests/sles4sap/qesapdeployment/configure.pm:37]: Please use setting name following naming convention `QESAPDEPLOY`

```
get_required_var('QESAPDEPLOY_USE_SAPCONF');
```

I try to follow this convention :
 - `QESAPDEPLOY_`  for settings used in test modules that belong to qesap regression
 - `QESAP_` for settings used in `lib/qesapdeployment.pm`

File: tests/sles4sap/qesapdeployment/configure.pm [hunk 2]
```diff
@@ -34,6 +34,7 @@ sub run {
     }
     $variables{OS_OWNER} = get_var('QESAP_CLUSTER_OS_OWNER', 'amazon') if check_var('PUBLIC_CLOUD_PROVIDER', 'EC2');
```
Comment (mpagot) [inline tests/sles4sap/qesapdeployment/configure.pm:35]: I'll fix this one

File: data/sles4sap/qe_sap_deployment/qesap_azure_uri.yaml [hunk 3]
```diff
@@ -29,7 +29,7 @@ ansible:
   create:
     - registration.yaml -e reg_code=${REG_CODE} -e email_address=${EMAIL}
     - pre-cluster.yaml
-    - sap-hana-preconfigure.yaml
+    - sap-hana-preconfigure.yaml -e use_sapconf=${SAPCONF}
```
[the remark this answers is not in the document]
Comment (lilyeyes) [reply data/sles4sap/qe_sap_deployment/qesap_azure_uri.yaml:32]: I just follow the coding style of `- registration.yaml -e reg_code=${REG_CODE} -e email_address=${EMAIL}`

File: tests/sles4sap/qesapdeployment/configure.pm (hunk 1 above)
Comment (lilyeyes) [reply tests/sles4sap/qesapdeployment/configure.pm:37]: Good suggestion. I will follow it next time when introducing new vars.
In this PR, the `USE_SAPCONF` is an old variable so I just use it.

**Comment by lilyeyes** [comment]
> For https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/data/sles4sap/qe_sap_deployment/qesap_azure.yaml and https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/data/sles4sap/qe_sap_deployment/qesap_azure_sapconf.yaml we used a different approach. We decided to duplicate the files. 

>>  Yes, agree, it duplicated the files. I noticed this approach but I think it duplicated files so I did not chose this way.

The idea there was linked to the qesap regression test purpose that is not to test a specific OS image but to test the `qe-sap-deployment` Terraform/Ansible code (and the relates openQA test wrapper https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/qesapdeployment.pm). So `qesap_azure.yaml ` was with no `-e use_sapconf=` at all also to probe the default in Ansible.
> 
> If you think that test used by `qesap_azure_uri.yaml` deviates from the original qesap regression purpose (and maybe it is the case) it is fine to have only one config.yaml but parametrized (as proposed in this PR). Otherwise we can duplicate `qesap_azure_uri.yaml` to `qesap_azure_uri_sapconf.yaml` .

At the beginning I used `qesap_azure.yaml` but test case failed so I used `qesap_azure_uri.yaml` instead.

**Pull request merged**